### ISSUE TITLE
Allow `private_dns_name` to be set via `aws_vpc_endpoint_service` resource

### ISSUE THREAD

**Issue by breathingdust** [hashicorp/terraform-provider-aws#16568]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

`private_dns_name` is currently an exported attribute, but the API does allow it to be set when creating a VPC Endpoint Service Configuration.

### New or Affected Resource(s)

* aws_vpc_endpoint_service

### Potential Terraform Configuration

```hcl
resource "aws_vpc_endpoint_service" "test" {
  acceptance_required = false
  network_load_balancer_arns = [
    aws_lb.test1.arn,
  ]
  private_dns_name = ""
}
```

### References

* https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateVpcEndpointServiceConfiguration.html
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/vpc_endpoint_service#private_dns_name

**Comment by ewbankkit** [comment]
Addressed via https://github.com/hashicorp/terraform-provider-aws/pull/16495.

**Comment by bflad** [comment]
Support for configurable `private_dns_name` and adding a `private_dns_name_configuration` attribute in the `aws_vpc_endpoint_service` resource has been merged and will release with version 3.23.0 of the Terraform AWS Provider. Thanks to @craigedmunds and @stijndehaes for their efforts here. 👍
